tests/foreman/cli/test_satellitesync.py

```diff
@@ -1139,6 +1139,91 @@ def test_positive_export_import_redhat_cv_with_huge_contents(
         for item in ['packages', 'source-rpms', 'package-groups', 'errata', 'module-streams']:
             assert exported_repo['content-counts'][item] == imported_repo['content-counts'][item]
 
+    @pytest.mark.tier2
+    def test_positive_export_cv_with_on_demand_repo(
+        self, export_import_cleanup_function, target_sat, module_org
+    ):
+        """Exporting CV version skips on_demand repo
+
+        :id: c366ace5-1fde-4ae7-9e84-afe58c06c0ca
+
+        :steps:
+
+            1. Create product
+            2. Create repos with immediate and on_demand download policy
+            3. Sync the repositories
+            4. Create CV with above product and publish
+            5. Attempt to export CV version contents
+
+        :expectedresults:
+
+            1. Export passes with warning and skips the on_demand repo
+
+        """
+        # Create custom product
+        product = make_product({'organization-id': module_org.id, 'name': gen_string('alpha')})
+
+        # Create repositories
+        repo_ondemand = make_repository(
+            {
+                'download-policy': 'on_demand',
+                'organization-id': module_org.id,
+                'product-id': product['id'],
+            }
+        )
+        repo_immediate = make_repository(
+            {
+                'download-policy': 'immediate',
+                'organization-id': module_org.id,
+                'product-id': product['id'],
+            }
+        )
+
+        # Sync repositories
+        Repository.synchronize({'id': repo_ondemand['id']})
+        Repository.synchronize({'id': repo_immediate['id']})
+
+        # Create cv and publish
+        cv = make_content_view({'name': gen_string('alpha'), 'organization-id': module_org.id})
+        ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo_ondemand['id'],
+            }
+        )
+        ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo_immediate['id'],
+            }
+        )
+        ContentView.publish({'id': cv['id']})
+
+        # Verify export directory is empty
+        assert validate_filepath(target_sat, module_org) == ''
+
+        # Export Content View version
+        result = ContentExport.completeVersion(
+            {'id': cv['id'], 'organization-id': module_org.id},
+            output_format='base',  # json output can't be parsed - BZ#1998626
+        )
+
+        # Warning is shown
+        assert (
+            """NOTE: Unable to fully export this version because it contains repositories """
+            """without the 'immediate' download policy. Update the download policy and sync """
+            """affected repositories. Once synced republish the content view and export the """
+            """generated version."""
+        ) in result
```

Comment:
Does this affect result.stdout? Also this assert will be fragile to future changes in wording of the warning. Maybe just checking for the first sentence would do, non blocking suggestion

Reply:
talked to rmynar offline, warning is shown with the successful export as well, so checking stdout wouldn't make a difference here